lsp/CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+# Contributing
+
+How to contribute.
+
+## Pre-Requisites + Initial Setup
+
+* `node` version 18+
+* `npm`
+* VSCode (recommended)
+
+Run:
+```
+git clone git@github.com:aws/aws-toolkit-common.git && \
+
+cd aws-toolkit-common && \
+
+git fetch origin test-typescript-lsp && git checkout test-typescript-lsp && \
+
+npm install
+```
+
+## Building The Language Server
+
+This project currently supports building a standalone language server through [pkg](https://github.com/vercel/pkg). pkg takes the project and packages it into an executable alongside a node binary.
+
+In order to build the standalone language server first install the pkg cli
+```bash
+npm install pkg -g
+```
+
+Now you have the ability to create standalone language server executables depending on the arguments you pass into pkg.
+
+If you want to create windows-x64, macos-x64, linux-x64 binaries you can use:
+```bash
+pkg .
+```
+
+to create a standalone executable for node16 for windows on arm you can do
+```bash
+pkg --targets node16-windows-arm64 .
+```
+
+to ensure the standalone language server is compressed even more you can do:
+```bash
+pkg --compress GZip .
+```
+
+## Running + Debugging
+
+> **NOTE**: Ensure your VSCode workspace is `aws-toolkit-common/lsp` or else certain functionality may not work.
+
+### With Minimal VSCode Client
+
+This repo contains a minimal vscode client that can be used to easily run and
+debug changes to this language server.
+
+1. In the `Run & Debug` menu, run `"Launch as VSCode Extension + Debugging"`
+2. Set breakpoints in `src` where needed.
+
+### With VSCode Toolkit Extension
```

Comment:
Cool, should be easy to simplify this even more once we start writing the packaging/distribution stuff